File: frontend/src/services/goodsService.js
import {postRequest} from "../utils/ajax";

export const getAllGoods = (data,callback) => {
    const url = `http://localhost:8080/getAllGoods`;
    postRequest(url, data, callback);
};

export const getGoodsByGoodsId = (data,callback) => {
    const url = `http://localhost:8080/getGoodsByGoodsId`;
    postRequest(url, data, callback);
};

export const getGoodsByGoodsType = (data,callback) => {
    const url = `http://localhost:8080/getGoodsByGoodsType`;
    postRequest(url, data, callback);
};

export const getGoodsByName = (data,callback) => {
    const url = `http://localhost:8080/getGoodsByName`;
    postRequest(url, data, callback);
};

export const getPopularGoods = (data,callback) => {
    const url = `http://localhost:8080/getPopularGoods`;
    postRequest(url, data, callback);
};

export const getAllAuctions = (data,callback) =>{
    const url = `http://localhost:8080/getAllAuctions`;
    postRequest(url,data,callback);
}

export const getAuctionByAuctionId = (data,callback) =>{
    const url = `http://localhost:8080/getAuctionByAuctionId`;
    postRequest(url,data,callback);
}

export const updateAuction = (data,callback) =>{
    const url = `http://localhost:8080/updateAuction`;
    postRequest(url,data,callback);
}

export const deleteGoodsByGoodsId = (data,callback) =>{
    const url = `http://localhost:8080/deleteGoodsByGoodsId`;
    postRequest(url,data,callback);
}

export const addAuction = (data,callback) => {
    const url = `http://localhost:8080/addAuction`;
    postRequest(url,data,callback);
}

export const editGoods = (data,callback) => {
    const url = `http://localhost:8080/editGoods`;
    postRequest(url,data,callback);
}

export const deleteAuctionByAuctionId = (data,callback) => {
    const url = `http://localhost:8080/deleteAuctionByAuctionId`;
    postRequest(url,data,callback);
}

export const editAuction = (data,callback) => {
    const url = `http://localhost:8080/editAuction`;
    postRequest(url,data,callback);
}

File: add_get/backend/backend/src/main/java/com/se128/jupiter/service/GoodsService.java
package com.se128.jupiter.service;

import com.se128.jupiter.entity.Auction;
import com.se128.jupiter.entity.Goods;
import org.springframework.data.domain.Page;

import java.util.List;

public interface GoodsService {

    Goods getGoodsByGoodsId(Integer goodsId);

    Goods editGoods(Goods goods);

    Goods addGoods(Goods goods);

    void deleteGoodsByGoodsId(Integer goodsId);

    List<Goods> getGoodsByGoodsType(Integer goodsType);

    List<Goods> getGoodsByName(String name);

    List<Goods> getGoodsByPageId(Integer pageId);

    Page<Goods> getAllGoods(Integer pageId, Integer pageSize, Integer goodsType);

    void saveViewCounter();

    List<Goods> getPopularGoods(Integer number, Integer goodsType);

    List<Auction> getAllAuctions();

    Auction getAuctionByAuctionId(Integer auctionId);

    Auction updateAuction(Integer auctionId, Integer userId, Double offer);

    List<Goods> getRecommendGoodsByUserId(Integer userId, Integer number);

    List<Goods> getRecommendGoodsInAll(Integer number);


    void deleteAuctionByAuctionId(Integer auctionId);

    Auction editAuction(Auction auction, Integer detailId, Integer goodsId);

    Auction addAuction(Auction auction, Integer goodsId, Integer detailId);
}
